csm/plugins/cortx/s3.py

```diff
@@ -37,88 +41,285 @@
+        """
+        Initialize S3RespParser.
+
+        :param for_list: list of keys that values should be treated as lists.
+        """
+        self._stack = []
+        self._force_list = force_list
+
+    def startElement(self, name: str, attrs: List[str]) -> None:
+        """
+        Implement startElement for xml.sax.ContentHandler.
 
-    def _create_boto_connection_object(self, **kwargs):
-        raise NotImplementedError
+        :param name: name of the element (e.g. for <Response> it's 'Response').
+        :param attrs: XML attributes of the element (e.g. <Response attr1='attr1)).
+                      attrs are ignored by the current implementation (S3 server doesn't use them).
+        """
+
+        e = Element(TYPE_KEY, name)
+        self._stack.append(e)
 
-    def _create_boto_connection(self, access_key, secret_key, config: S3ConnectionConfig,
-                                session_token=None):
+    def endElement(self, name: str) -> None:
         """
-        Helper function that creates IAM connection for the given credentials and configuration
-        :returns: an IAMConnection object
+        Implement endElement for xml.sax.ContentHandler.
+
+        :param name: name of the element (e.g. for </Response> it's 'Response').
         """
 
-        # Here we're overwriting the global boto config parameters.
-        # In our case, these parameters are only important at the IAMConnection creation
-        # So it will not affect other connections
-        ca_cert = DEFAULT_CA_CERTS_FILE
-        if config.ca_cert_file:
-            ca_cert = config.ca_cert_file
+        acc = None
+        # Logic: extract values from stack and merge them until the key appears
+        # If key's name matches the name from parameters assign the merged value
+        # to the key and push the resulting value to the stack
+        while len(self._stack):
+            e = self._stack.pop()
+            if e.type == TYPE_KEY:
+                if e.val != name:
+                    raise CsmTypeError(f'Malformed XML: expected </{name}>, got </{e.val}>')
+                if name in self._force_list:
+                    acc = [acc]
+                acc_elem = Element(TYPE_VAL, {name: acc})
+                self._stack.append(acc_elem)
+                break
+            else:
+                acc = S3RespParser.merge(acc, e.val)
+        if not len(self._stack):
+            raise CsmTypeError(f'Malformed XML: unexpected </{name}>')
 
-        if not boto_config.has_section('Boto'):
-            boto_config.add_section('Boto')
+    def characters(self, content: str) -> None:
+        """
+        Implements characters for xml.sax.ContentHandler.
 
-        boto_config.set('Boto', 'ca_certificates_file', ca_cert)
+        :param content: text appeared during parsing an XML document.
+        """
+
+        content = content.strip()
+        if not content:
+            return
+        e = Element(TYPE_VAL, content)
```

Comment:
Instead of `e` please use a relevant variable name

Reply:
Are we seriously blocking a PR because a variable name in a 5-line function body is not "relevant"?

See http://catb.org/jargon/html/B/bikeshedding.html .

Reply:
@tb-s In the past similar kind of review comments were given by reviewers/architects for the variable names. I personally have corrected proper variable names just because the suggested was more relevant to that function/code.
3-4 days before I encountered a code that says `(a==b)`. If we allow 'e` then we have to allow all such occurrences. 
In one of the functions instead of `e` `element` is used as variable.

Reply:
@tb-s  however blocking this PR have multiple reasons like approval of getting new library added, and a thorough testing on a fresh deployed VM is required and tested by QA before we merge it,
 but this variable name not proper itself is more than enough to block the PR.

Reply:
Updated the name.

Reply:
@kupranay , Reader can't treat `e` like `event` for one reason: right part of assignment named `Element`. Such code review requests are overhead here.

Reply:
I removed the whole parser since Ujjwal approved xmltodict as a dependency.

Reply:
@dmitrydb  code readability as important as much as code correctness. I dont think this is overhead. This is good coding practice.